backend/core/src/main/java/com/feed_grabber/core/request/RequestService.java

```diff
@@ -193,6 +205,39 @@ public Date closeNow(UUID requestId) throws NotFoundException {
         return closeDate;
     }
 
+    public Date close(UUID requestId) throws NotFoundException {
+        var request = requestRepository.findById(requestId)
+                .orElseThrow(() -> new NotFoundException("Request not found"));
+        if (request.getCloseDate() != null) {
+            return new Date();
+        }
```

Comment:
Are you shure about  return new Date(); Maybe return request.getCloseDate()?

Reply:
Year, you right.